### PR TITLE
Updated Travis config to enable running on Mac and also fixed test

### PR DESCRIPTION
**Pull request by kichalla**
@pranavkm @cesarbs

### CODE REVIEW

File: samples/DeveloperExceptionPageSample/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (pranavkm) [inline samples/DeveloperExceptionPageSample/Properties/launchSettings.json]: Remove. Add to .gitignore.

**Comment by kichalla** [comment]
:up: :date:

**Comment by pranavkm** [comment]
cc @NTaylorMullen

**Comment by pranavkm** [comment]
:shipit:  Please get sign off from @NTaylorMullen for travis changes.

File: .travis.yml [hunk 2]
```diff
@@ -10,9 +10,14 @@ addons:
     - libssl-dev
     - libunwind8
     - zlib1g
+before_install:
+  - if test "$TRAVIS_OS_NAME" == "osx"; then brew update; brew install icu4c; fi
 env:
   - KOREBUILD_DNU_RESTORE_CORECLR=true KOREBUILD_TEST_DNXCORE=true
```
Comment (kichalla) [inline .travis.yml]: @NTaylorMullen Razor repo's travis.yml has the following setting `MONO_THREADS_PER_CPU=2000`, should I be using it here too or does it not matter anymore as I guess `KOREBUILD_DNU_RESTORE_CORECLR` is to say that restore is done using coreclr dnx?

File: test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public static TheoryData RelativePathsData
                     "TestFiles/SourceFile.txt"
                 };
 
-                if (!TestPlatformHelper.IsLinux)
+                if (!(TestPlatformHelper.IsLinux || TestPlatformHelper.IsMac))
```
Comment (dougbu) [inline test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs]: Suggest flipping this to `if (TestPlatformHelper.IsWindows)`

File: .travis.yml (hunk 2 above)
Comment (dougbu) [inline .travis.yml]: @kichalla you're correct that `MONO_THREADS_PER_CPU` doesn't matter for `dnu restore` anymore. Only need to set this in `.travis.yml` if testing sometimes hits threading issues. Fairly obvious that's not the case here -- tests haven't been terrible.

**Comment by dougbu** [comment]
:shipit: if Travis builds succeed

File: test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs (hunk 3 above)
Comment (kichalla) [inline test/Microsoft.AspNet.Diagnostics.Tests/DeveloperExceptionPageMiddlewareTest.cs]: Yeah I contemplated doing that but I recall someone mentioning that this is not preferred in the scenario where there would be support for more platforms in future and we would know that certain tests fail on them.

**Comment by NTaylorMullen** [comment]
:shipit:

**Comment by kichalla** [comment]
573bdcc318042afd70ce94fc2828220ece130f05

**Pull request closed without merging**